app/assets/stylesheets/content/_tooltips.sass

```diff
@@ -26,27 +26,159 @@
+  &:before, &:after
+    bottom: 50%
+  &:before
+    margin-bottom: $tooltip--arrow-size * -1 + 1
+  &:after
+    margin-bottom: $tooltip--height/-1.5
+
+  &.-multiline
+    &:before, &:after
+      transform: translateY(50%)
+      margin-bottom: 0
+
+@mixin tooltip--colorize($color: $tooltip--background-color, $textColor:$tooltip--font-color)
+  @each $position in top, right, bottom, left
+    &.tooltip--#{$position}
+      &:before
+        border-#{$position}-color: $color
+      &:after
+        background-color: $tooltip--background-color
+        color: $tooltip--font-color
+
+[data-tooltip]
   position: relative
-  z-index: 24
-  &.hover, &:hover
-    z-index: 25
-    color: #000
-  span.tip
-    display: none
-    text-align: left
-
-div.tooltip
-  &:hover span.tip, &.hover span.tip
-    display: block
+  display: inline-block
+  box-sizing: content-box
+  // TODO: this "fixes" the spacing issue in text, we need to find a better way for that
+  padding-right: 5px
+  +tooltip--colorize
+
+  &:before, &:after
     position: absolute
-    top: 12px
-    left: 24px
-    width: 270px
-    border: 1px solid #555
-    background-color: #fff
-    padding: 4px
-    font-size: 0.8em
-    color: #505050
+    visibility: hidden
+    opacity: 0
+    z-index: 99999
+    box-sizing: content-box
+    transform: translate3d(0, 0, 0)
+
+  &:before
+    content: ''
+    border: $tooltip--arrow-size solid transparent
+
+  &:after
+    height: $tooltip--height
+    padding: $tooltip--height/2 $tooltip--height/2 0 $tooltip--height/2
+    font-size: 13px
+    line-height: $tooltip--height/2
+    white-space: nowrap
+    content: attr(data-tooltip)
+    border: $tooltip--border
+
+  &:hover, &:focus
+    background-color: transparent
+    &::before, &::after
+      opacity: 1
+      visibility: visible
+
+  &.-multiline
+    &:after
+      height: auto
+      width: 150px
+      padding: $tooltip--height/2
+      line-height: $tooltip--height - 3px
+      white-space: normal
+      text-align: left
+
+  &.tooltip--right
+    @extend %tooltip--left-right
+    &:before, &:after
+      left: 100%
+    &:before
+      margin-left: -2px
+    &:after
+      margin-left: 10px
+
+  &.tooltip--left
+    @extend %tooltip--left-right
+    &:before, &:after
+      right: 100%
+    &:before
+      margin-right: -2px
+    &:after
+      margin-right: 10px
+
+  &.tooltip--top
+    @extend %tooltip--top-bottom
+    &:before, &:after
+      bottom: 100%
+    &:before
+      margin-bottom: $tooltip--arrow-size * -1 + 1
+    &:after
+      margin-bottom: 7px
+    &:hover
+      &:before, &:after
+        transform: translate(-50%, 0px)
+
+  &.tooltip--bottom
+    @extend %tooltip--top-bottom
+    &:before, &:after
+      top: 100%
+    &:before
+      margin-top: $tooltip--arrow-size * -1 + 1
+    &:after
+      margin-top: 7px
+    &:hover
+      &:before, &:after
+        transform: translate(-50%, 0)
+
+// tooltips for table content (TODO: legacy?)
+table
+  .tooltip
```

Comment:
@floriank just found _yet another tooltip implementation_ in the timelines view
… (it wouldn't be OpenProject if there weren't at least three implementations™ of the same thing).

![seeded_project_-_timelines__ganz_simple_-_openproject](https://cloud.githubusercontent.com/assets/755/7083199/3e7e2ef4-df5f-11e4-9072-1511feb4361d.png)
`.tl-tooltip` and related classes in `timelines.css.sass`

In retrospect, we might actually have been able to reuse some of this implementation (CSS, at least) for this WP.